### PR TITLE
feat(expo-cli): ask to close expo when upgrading

### LINKED ISSUE #1108: Warn if server is running when you run upgrade command and prompt to close it
The first step in "Upgrading Your App" in the [release notes](https://blog.expo.io/expo-sdk-35-is-now-available-beee0dfafbf4) is to "Close your Expo CLI server". We should get rid of this by instead checking if a server is running for the project and asking the user if they want us to close it. If they don't., then we stop the upgrade.

### PR DESCRIPTION
**Pull request by byCedric**
Fixes #1108

### CODE REVIEW

**Comment by byCedric** [comment]
Just a random side note, but it seems that Circle isn't running the tests for non-expo-members. I mean, I'm happy to contribute but not running those tests "kind of" makes me a bit nervous 😅

File: packages/expo-cli/src/commands/upgrade.ts [hunk 1]
```diff
@@ -147,6 +147,23 @@ async function upgradeAsync(requestedSdkVersion: string | null, options: Options
...
+  let status = await Project.currentStatus(projectRoot);
+  if (status === 'running') {
+    let answer = await prompt({
+      type: 'confirm',
+      name: 'closeExpo',
+      message: 'Expo needs to close before we can upgrade your dependencies. Do you want to close it?',
+    });
+
+    if (!answer.closeExpo) {
+      return;
```
Comment (fson) [inline packages/expo-cli/src/commands/upgrade.ts]: In this case the exit status of the command should be non-zero. You can, for example, throw a `CommandError` to achieve this.

Comment (byCedric) [reply packages/expo-cli/src/commands/upgrade.ts]: Thanks! Will do in about 3-4 hours!

Comment (byCedric) [reply packages/expo-cli/src/commands/upgrade.ts]: I added it, not sure if it has the proper name though, maybe `BAD_CHOICE` is also a good one? I didn't add it to the [error codes](https://github.com/expo/expo-cli/blob/master/packages/expo-cli/src/CommandError.ts#L5) because the other two command errors are not listed there. But I'd be happy to add it if necessary.

File: packages/expo-cli/src/commands/upgrade.ts [hunk 2]
```diff
@@ -147,6 +147,23 @@ async function upgradeAsync(requestedSdkVersion: string | null, options: Options
     }
   }
 
+  // Can't upgrade if Expo is running
+  let status = await Project.currentStatus(projectRoot);
+  if (status === 'running') {
+    let answer = await prompt({
+      type: 'confirm',
+      name: 'closeExpo',
+      message: 'Expo needs to close before we can upgrade your dependencies. Do you want to close it?',
```
Comment (brentvatne) [inline packages/expo-cli/src/commands/upgrade.ts]:
```suggestion
      message: 'We found an existing expo-cli instance running for this project, it needs to be closed to continue.',
```

maybe something like this and then present two options:

```
- Close it for me and continue upgrade
- Cancel, I'm not ready yet
```

tbh i'm not sure how important this is though.. we could just close it for people and tell them that we closed it. it's not a destructive action, it costs nothing for them to close and reopen the packager.

Comment (byCedric) [reply packages/expo-cli/src/commands/upgrade.ts]: I don't think a lot of devs will even encounter this. Maybe just telling them that we closed it is sufficient and leaves some extra room for other questions it might encounter in the future 😄

Comment (brentvatne) [reply packages/expo-cli/src/commands/upgrade.ts]: sounds good

Comment (brentvatne) [reply packages/expo-cli/src/commands/upgrade.ts]: approved so we can ship it once you have made this change!

File: packages/expo-cli/src/commands/upgrade.ts (hunk 1 above)
Comment (fson) [reply packages/expo-cli/src/commands/upgrade.ts]: The error code doesn't matter all that much in this case – it's mostly useful for cases where we want to catch a specific error, e.g. a common pattern is like this:
```
try {
  doStuffThatMightFail();
} catch (error) {
  if (error.code === 'BAD_CHOICE') {
    trySomethingElse();
  } else {
    throw error;
  }
}
```
Tl;dr: `BAD_CHOICE` is a good choice! 😆

Comment (byCedric) [reply packages/expo-cli/src/commands/upgrade.ts]: Thanks Ville! ❤️  I'll remember that next time 😄 

For now, I think this specific instance is superseded by Brent's suggestion to just tell instead of ask, right?

File: packages/expo-cli/src/commands/upgrade.ts (hunk 2 above)
Comment (byCedric) [reply packages/expo-cli/src/commands/upgrade.ts]: Done! 😄

File: packages/expo-cli/src/commands/upgrade.ts [hunk 3]
```diff
@@ -147,6 +147,15 @@ async function upgradeAsync(requestedSdkVersion: string | null, options: Options
     }
   }
 
+  // Can't upgrade if Expo is running
+  let status = await Project.currentStatus(projectRoot);
+  console.log({ status });
```
Comment (brentvatne) [inline packages/expo-cli/src/commands/upgrade.ts]:
```suggestion
```

Comment (byCedric) [reply packages/expo-cli/src/commands/upgrade.ts]: My bad, i'll remove it right away

**Comment by byCedric** [comment]
My bad, it's gone and won't be back!
